python/postprocessing/modules/jme/fatJetUncertainties.py

```diff
@@ -547,8 +547,8 @@ def analyze(self, event):
                     genGroomedSubJets = None
                     genGroomedJet = None
                 if jet.subJetIdx1 >= 0 and jet.subJetIdx2 >= 0:
-                    groomedP4 = subJets[jet.subJetIdx1].p4() + subJets[
-                        jet.subJetIdx2].p4()  # check subjet jecs
+                    groomedP4 = (subJets[jet.subJetIdx1].p4() * (1. - subJets[jet.subJetIdx1].rawFactor)) \
+                        + (subJets[jet.subJetIdx2].p4() * (1. - subJets[jet.subJetIdx2].rawFactor))
```

Comment:
Can you check that the eta and phi of the jet are not affected? The raw factor should only be applied to the pt and mass.

Reply:
@camclean Are you referring to eta and phi of the AK8 parent jet or each of the subjets?

Reply:
The one you get from groomedP4

Reply:
The attached slide shows pT, eta and phi distributions of subjet 1 (top) and subjet 2 (bottom). Black solid histogram is with JEC applied (by default in NanoAOD) and blue dashed histogram is without the JEC applied (as implemented by the fixed in this PR). The implementation does not affect the subjets' eta and phi variables.

[SubJetKinematics_WithAndWithoutJEC.pdf](https://github.com/cms-nanoAOD/nanoAOD-tools/files/6593069/SubJetKinematics_WithAndWithoutJEC.pdf)